hooks/node_handlers/base_export_handler.py

```diff
@@ -0,0 +1,644 @@
+        else:
+            # Lets clear these as we can't have just location and variation without element
+            sgtk_location.set("")
+            sgtk_variation.set("")
+
+        location = sgtk_location.evalAsString().strip() or None
+        fields["location"] = location
+
+        variation = sgtk_variation.evalAsString().strip() or None
+        fields["variation"] = variation
+
+    def _update_optional_keys(self, node, template, fields):
+        """
+        Update the optional keys parameter to include the currently 'in use'
+        fields and their values.
+
+        :param node: A :class:`hou.Node` instance.
+        :param template: An :class:`sgtk.Template`.
+        :param dict fields: The template fields.
+        """
+        optional_fields = {}
+        for key_name in self.get_optional_keys(template):
+            field = fields.get(key_name)
+            if field:
+                optional_fields[key_name] = field
+        parm = node.parm(self.OPTIONAL_KEYS)
+        if parm:
+            parm.set(json.dumps(optional_fields))
+
+    def _update_all_versions(self, node, all_versions):
+        """
+        Update all the versions that exist on disk.
+
+        :param node: A :class:`hou.Node` instance.
+        :param list(int) all_versions: Updated list of all versions.
+        """
+        if all_versions != self._get_all_versions(node):
+            all_versions_str = ",".join(map(str, all_versions))
+            sgtk_all_versions = node.parm(self.SGTK_ALL_VERSIONS)
+            sgtk_all_versions.set(all_versions_str)
+            
+    def _refresh_file_path(self, node, update_version=True):
+        """
+        Refresh the file paths generated by the node handler.
+
+        :param node: A :class:`hou.Node` instance.
+        :param bool update_version: Update the version drop down accordingly.
+        """
+        output_parm = node.parm(self.OUTPUT_PARM)
+
+        context = self.parent.context
+        template = self.get_work_template(node)
+        fields = context.as_template_fields(template, validate=True)
+
+        self._update_template_fields(node, fields)
+        self._update_optional_keys(node, template, fields)
+        
+        all_versions = self._resolve_all_versions_from_fields(node, fields, template)
+        sgtk_version = node.parm(self.SGTK_VERSION)
+        self._update_all_versions(node, all_versions)
+        
+        if update_version:
+            sgtk_version.set(len(all_versions))
+        current = sgtk_version.evalAsString()
+        resolved_version = self._resolve_version(all_versions, current)
+
+        sgtk_resolved_version = node.parm(self.SGTK_RESOLVED_VERSION)
+        sgtk_resolved_version.set(str(resolved_version))
+        
+        fields["version"] = resolved_version
+        try:
+            new_path = self.get_work_template(node).apply_fields(fields)
+        except sgtk.TankError:
+            new_path = self.DEFAULT_ERROR_STRING
```

Comment:
Add extra error context info to log
```suggestion
            new_path = self.DEFAULT_ERROR_STRING
            self.logger.exception('Failed to calculate path for "%s"', node)
```